app/sweepers.py

```diff
@@ -0,0 +1,76 @@
+
+"""
+This module contains the definitions of Sagittariidae's sweeper processes.
+These are asynchronous processes that designed to drive the current state of
+the world to a desired final state.
+
+The sweepers are run as processes separate from the webservice.  They may be
+invoked by name by executing this module, typically from a scheduler such as
+`cron`.
+"""
+
+import os
+import shutil
+import sys
+
+import app as sagittariidae
+import models
+
+
+logger = sagittariidae.app.logger
+
+class Sweeper(object):
+
+    def run(self):
+        raise NotImplementedError()
+
+    def sweep(self):
+        try:
+            self.run()
+        except Exception, e:
+            logger.error('Unhandled exception in sweeper %s', self, exc_info=e)
```

Comment:
Good point. And failing to set `FileStatus.complete`, the file would still show up red in the UI. Would the user be able to reupload that file, or would it then violate the unique filename criterion?